### PR TITLE
feat: ensures vpcId is recalled (if available) on account_activate

### PR DESCRIPTION
**Pull request by OhCloud**
Issue #, if available:

**Description of Changes**
Previously, `account activate` did not recall any existing vpcIds and users had to add it in manually. Now, if a vpcId exists, it will recall and use it in-place of generating a new one. 

[//]: #  (A description of the change that you made and the new user experience that it creates)

**Description of how you validated changes**
Added unit tests as well as linted using `make`. 

[//]: #  (A description of you validated your changes and any relevant logs that show your change works)
Screen Shot 2022-03-08 at 2 09 08 PM

**Checklist**

- [ ] If this change would make any existing documentation invalid, I have included those updates within this PR
- [x] I have added unit tests that prove my fix is effective or that my feature works
- [x] I have linted my code before raising the PR
- [x] Title of this Pull Request follows Conventional Commits standard: https://www.conventionalcommits.org/en/v1.0.0/

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/cdk/lib/stacks/core-stack.ts [hunk 1]
```diff
@@ -82,6 +82,9 @@ export class CoreStack extends Stack {
       },
     });
 
+    console.log("context passed in App :point_right:", this.node.tryGetContext("fromApp"));
```
Comment (tneely) [inline packages/cdk/lib/stacks/core-stack.ts:85]: Can you remove this log statement?

File: packages/cli/internal/pkg/cli/account_activate.go [hunk 2]
```diff
@@ -33,10 +35,11 @@ A new VPC will be created if not specified.`
...
 otherwise the parsing will fail.`
-	cdkCoreDir   = ".agc/cdk/apps/core"
-	bucketPrefix = "agc"
-	activateKey  = "activate"
-	bootstrapKey = "bootstrap"
+	cdkCoreDir    = ".agc/cdk/apps/core"
+	bucketPrefix  = "agc"
+	activateKey   = "activate"
+	bootstrapKey  = "bootstrap"
+	coreStackName = "Agc-Core"
```
Comment (tneely) [inline packages/cli/internal/pkg/cli/account_activate.go:42]: Can you move this to [awsresources.go](https://github.com/aws/amazon-genomics-cli/blob/main/packages/cli/internal/pkg/cli/awsresources/awsresources.go) as a `RenderCoreStackName()` function?

File: packages/cli/internal/pkg/cli/account_activate.go [hunk 3]
```diff
@@ -61,62 +65,97 @@ func newAccountActivateOpts(vars accountActivateVars) (*accountActivateOpts, err
...
 	}
 
 	environmentVars := []string{
-		fmt.Sprintf("%s=%s", constants.AgcBucketNameEnvKey, o.bucketName),
-		fmt.Sprintf("%s=%t", constants.CreateBucketEnvKey, !exists),
-		fmt.Sprintf("%s=%s", constants.AgcVersionEnvKey, version.Version),
-	}
-	if o.vpcId != "" {
-		environmentVars = append(environmentVars, fmt.Sprintf("%s=%s", constants.VpcIdEnvKey, o.vpcId))
+		fmt.Sprintf("AGC_BUCKET_NAME=%s", o.bucketName),
```
Comment (tneely) [inline packages/cli/internal/pkg/cli/account_activate.go:118]: Why are you no longer using the constants?

File: packages/cli/internal/pkg/cli/account_activate.go [hunk 4]
```diff
@@ -61,62 +65,97 @@ func newAccountActivateOpts(vars accountActivateVars) (*accountActivateOpts, err
...
-	account, err := o.stsClient.GetAccount()
-	if err != nil {
-		return "", err
+func (o accountActivateOpts) getVpcId() (string, error) {
+	if o.vpcId != "" {
+		return o.vpcId, nil
+	} else {
+		stackOutputs, err := o.cfnClient.GetStackOutputs(coreStackName)
+		if err != nil {
+			if err == cfn.StackDoesNotExistError {
```
Comment (tneely) [inline packages/cli/internal/pkg/cli/account_activate.go:147]: Go error handling can be weird. I think what you have works but can you use `errors.Is(err, cfn.StackDoesNotExistError)` instead?

File: packages/cdk/lib/stacks/core-stack.ts (hunk 1 above)
Comment (OhCloud) [reply packages/cdk/lib/stacks/core-stack.ts:85]: np.

File: packages/cli/internal/pkg/cli/account_activate.go [hunk 5]
```diff
@@ -61,62 +65,97 @@ func newAccountActivateOpts(vars accountActivateVars) (*accountActivateOpts, err
...
+
+func (o accountActivateOpts) generateDefaultBucket() (string, error) {
+	account, err := o.stsClient.GetAccount()
+	if err != nil {
+		return "", err
+	}
+	return generateBucketName(account, o.region), nil
+}
+
+func (o accountActivateOpts) generateEnvVars() ([]string, error) {
```
Comment (a-li) [inline packages/cli/internal/pkg/cli/account_activate.go:103]: I understand that you've created this function as `Execute()` is getting too big. Although you're catching the err on line 97, the err handling flow has changed and the program now wait until all these checks are complete as opposed to original fail-fast which is a better mechanism in my opinion. 

Let me know if I'm missing something.

**Comment by a-li** [comment]
@OhCloud Consider updating this with current behavior: https://github.com/aws/amazon-genomics-cli/blob/main/site/content/en/docs/Concepts/accounts.md?plain=1#L42

File: packages/cli/internal/pkg/cli/account_activate.go (hunk 3 above)
Comment (OhCloud) [reply packages/cli/internal/pkg/cli/account_activate.go:118]: this was an odd import error. constants have now been restored.

File: packages/cli/internal/pkg/cli/account_activate.go (hunk 5 above)
Comment (elliot-smith) [reply packages/cli/internal/pkg/cli/account_activate.go:104]: Hey Angela,
If you look through the function, we exit out of the fail whenever the function returns an error so logically the code is exactly the same but it is easier to read. 
I.e. the error from line 97 is passed to line 108 which is then passed to line 79. The original flow would have been 97 to line 79.

Comment (a-li) [reply packages/cli/internal/pkg/cli/account_activate.go:104]: Thanks Elliot for the detail explanation. I read it more closely now. It is doing I was intended. No change is required here.

**Review by elliot-smith — DISMISSED** [review]

**Review by a-li — DISMISSED** [review]

File: packages/cli/internal/pkg/cli/account_activate.go [hunk 6]
```diff
@@ -147,6 +188,7 @@ func displayProgress(progressStream cdk.ProgressStream, displayMsg string) error
 			}
 			lastEvent = event
 		}
+		log.Info().Msg(strings.Join(lastEvent.Outputs, "\n"))
```
Comment (tneely) [inline packages/cli/internal/pkg/cli/account_activate.go:191]: I just want to verify this is an intended change and not some debugging log.

Comment (OhCloud) [reply packages/cli/internal/pkg/cli/account_activate.go:191]: running `account activate -v` will return a neat set of outputs including the TableName, VpcId, and Bucket. 
Screen Shot 2022-03-08 at 2 09 08 PM

**Review by tneely — APPROVED** [review]

**Pull request merged**